Sources/SwiftDocC/Model/Rendering/RenderNodeTranslator.swift

```diff
@@ -1852,23 +1852,50 @@ public struct RenderNodeTranslator: SemanticVisitor {
     private func variants(for documentationNode: DocumentationNode) -> [RenderNode.Variant] {
         let generator = PresentationURLGenerator(context: context, baseURL: bundle.baseURL)
         
-        return documentationNode.availableSourceLanguages
-            .sorted(by: { language1, language2 in
+        func mapSourceLanguageToVariants(identifier: ResolvedTopicReference, sourceLanguages: Set<SourceLanguage>) -> [(SourceLanguage, [ResolvedTopicReference])] {
```

Comment:
minor: I don't think this function is helping the readability of the rest of this code. 

----

The first usage could be written as one these alternatives without the `mapSourceLanguageToVariants` function
```swift
var allVariants: [SourceLanguage: ResolvedTopicReference] = documentationNode.availableSourceLanguages.reduce(into: [:]) { partialResult, language in
    partialResult[language] = identifier
}
```

```swift
var allVariants = [SourceLanguage: ResolvedTopicReference]()
for language in documentationNode.availableSourceLanguages {
    allVariants[language] = identifier
}
```

I find both alternatives to be more explicit in how the data is being built up. 

---

Similarly, the second usage could be written s one these alternatives without the `mapSourceLanguageToVariants` function

```swift
for language in alternateRepresentationReference.sourceLanguages.subtracting(allVariants.keys) {
    allVariants[language] = alternateRepresentationReference
}
```

```swift
allVariants.merge(
    alternateRepresentationReference.sourceLanguages.map { ($0, alternateRepresentationReference) }
) { existing, _ in existing }
```

---

Also, I don't see the reason to store an array of references for each language when the only value ever is a single element, so in the examples above I used `[SourceLanguage: ResolvedTopicReference]` instead of `[SourceLanguage: [ResolvedTopicReference]]`. 

The only difference that I can see is that the creation of the `RenderNode.Variant` towards the end would create the array inline from the only reference
```diff
  RenderNode.Variant(
      traits: [.interfaceLanguage(sourceLanguage.id)],
+     paths: [
-     paths: references.map { reference in
          generator.presentationURLForReference(reference).path
-     }
+     ]
  )

```

Reply:
Seems reasonable to me, I agree it makes it clearer to use a `ResolvedTopicReference` over a `[ResolvedTopicReference]` to emphasise there can only be one. The main reason I wasn't doing that originally is to remove the changes of unintentionally causing a regression in behaviour. 

Previously, we weren't enforcing having only one path per trait, so even though I can't think of a case where there would have been multiple paths per trait, it could have theoretically happened. I'm not sure if enforcing it will have unintentional consequences, but I think it does make it clearer to enforce it here.

Made the suggested changes, thanks for the code suggestions!